iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/ProjectResourceFactoryTest.java

```diff
@@ -237,4 +244,108 @@ void shouldReturnNullForOtherKinds() {
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideMapResource")
+  void shouldCreateMapResource(String kindName, Class<? extends MapResource> clazz) {
+    // language=yaml
+    var code = """
+      apiVersion: v1
+      kind: %s
+      metadata:
+        namespace: my-namespace
+      data:
+        key1: "value1"
+        key2: "value2"
+      """.formatted(kindName);
+    var tree = (MappingTree) PARSER.parse(code, null).documents().get(0);
+
+    var mapResource = (MapResource) ProjectResourceFactory.createResource("mapResource.yaml", tree);
+
+    assertThat(mapResource).isInstanceOf(clazz);
+    assertThat(mapResource.path()).isEqualTo("mapResource.yaml");
+    assertThat(mapResource.values()).containsKeys("key1", "key2");
+    assertThat(mapResource.values().get("key1"))
+      .isInstanceOf(ScalarTree.class)
+      .extracting(s -> ((ScalarTree) s).value())
+      .isEqualTo("value1");
+    assertThat(mapResource.values().get("key2"))
+      .isInstanceOf(ScalarTree.class)
+      .extracting(s -> ((ScalarTree) s).value())
+      .isEqualTo("value2");
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideMapResource")
+  void shouldParseMapResourceWithComplexValue(String kindName, Class<? extends MapResource> clazz) {
+    // language=yaml
+    var code = """
+      apiVersion: v1
+      kind: %s
+      metadata:
+        namespace: my-namespace
+      data:
+        key1:
+          subkey:
+            "value1"
+      """.formatted(kindName);
+    var tree = (MappingTree) PARSER.parse(code, null).documents().get(0);
+
+    var mapResource = (MapResource) ProjectResourceFactory.createResource("mapResource.yaml", tree);
+
+    assertThat(mapResource).isInstanceOf(clazz);
+    assertThat(mapResource.path()).isEqualTo("mapResource.yaml");
+    assertThat(mapResource.values()).containsKeys("key1");
+    assertThat(mapResource.values().get("key1"))
+      .isInstanceOf(MappingTree.class);
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideMapResource")
+  void shouldStayEmptyForIncorrectMapResourceDataFormat(String kindName, Class<? extends MapResource> clazz) {
+    // language=yaml
+    var code = """
+      apiVersion: v1
+      kind: %s
+      metadata:
+        namespace: my-namespace
+      data:
+      - key1: "value1"
+      """.formatted(kindName);
+    var tree = (MappingTree) PARSER.parse(code, null).documents().get(0);
+
+    var mapResource = (MapResource) ProjectResourceFactory.createResource("mapResource.yaml", tree);
+
+    assertThat(mapResource).isInstanceOf(clazz);
+    assertThat(mapResource.path()).isEqualTo("mapResource.yaml");
+    assertThat(mapResource.values()).isEmpty();
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideMapResource")
+  void shouldHandleKeyNotScalarInMapResource(String kindName, Class<? extends MapResource> clazz) {
+    // language=yaml
+    var code = """
+      apiVersion: v1
+      kind: %s
+      metadata:
+        namespace: my-namespace
+      data:
```

Comment:
Please add a test case where `data` is missing. The `data` is optional: https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-object